src/home_robot/home_robot/navigation_planner/discrete_planner.py

```diff
@@ -372,24 +437,57 @@ def _get_short_term_goal(
             # Set multi goal to the dilated goal map
             # We will now try to find a path to any of these spaces
             planner.set_multi_goal(dilated_goal_map, self.timestep)
+            # Set dilated goal map for use with simulation code - use this to compute closest goal
+            goal_map = dilated_goal_map
+            goal_distance_map, closest_goal_pt = self.get_closest_traversible_goal(
+                traversible, dilated_goal_map, start
+            )
         else:
-            navigable_goal = planner._find_nearest_to_multi_goal(goal_map)
-            navigable_goal_map = np.zeros_like(goal_map)
-            navigable_goal_map[navigable_goal[0], navigable_goal[1]] = 1
-            planner.set_multi_goal(navigable_goal_map, self.timestep)
+            navigable_goal_map = planner._find_within_distance_to_multi_goal(
+                goal_map, self.min_goal_distance_cm / self.map_resolution
+            )
+            if not np.any(navigable_goal_map):
+                replan = True
+            else:
+                planner.set_multi_goal(navigable_goal_map, self.timestep)
+            goal_distance_map, closest_goal_pt = self.get_closest_goal(goal_map, start)
```

Comment:
Yes, we can use `get_closest_goal` here in case we don't already have the `dilated goal` computed.

My change was using `get_closest_traversible_goal` here (with plan_to_dilated_goal=False). Dilating goal was necessary to ensure it comes in contact with other traversible regions. 

The current change should work fine though.

Reply:
i think for the dilated-goal planning code nothing should have changed

Reply:
You are right. I am still using `plan_to_dilated_goal=True`